CONTRIBUTING.markdown

```diff
@@ -1,106 +1,126 @@
+Whether you're a developer, a designer, or just a Jekyll devotee, there are lots of ways to contribute. Here's a few ideas:
 
-    $ script/cucumber features/blah.feature
+* [Install Jekyll on your computer](https://jekyllrb.com/docs/installation/) and kick the tires. Does it work? Does it do what you'd expect? If not, [open an issue](https://github.com/jekyll/jekyll/issues/new) and let us know.
+* Comment on some of the project's [open issue](https://github.com/jekyll/jekyll/issues). Have you experienced the same problem? Know a work around? Do you have a suggestion for how the feature could be better?
+* Read through [the documentation](http://jekyllrb.com/docs/home/), and click the "improve this page" button, any time you see something confusing, or have a suggestion for something that could be improved.
+* Browse through [the Jekyll discussion forum](https://talk.jekyllrb.com/), and lend a hand answering questions. There's a good chance you've already experienced what another user is experiencing.
+* Find [an open issue](https://github.com/jekyll/jekyll/issues) (especially [those labeled `help-wanted`](https://github.com/jekyll/jekyll/issues?q=is%3Aopen+is%3Aissue+label%3Ahelp-wanted)), and submit a proposed fix. If it's your first pull request, we promise we won't bite, and are glad to answer any questions.
+* Help evaluate [open pull requests](https://github.com/jekyll/jekyll/pulls), by testing the changes locally and reviewing what's proposed.
 
-Both `script/test` and `script/cucumber` can be run without arguments to
-run its entire respective suite.
+## Submitting a pull request
+
+### Pull requests generally
+
+* The smaller the proposed change, the better. If you'd like to propose two unrelated changes, submit two pull requests.
+
+* The more information, the better. Make judicious use of the pull request body. Describe what changes were made, why you made them, and what impact they will have for users.
+
+* Pull request are easy and fun. If this is your first pull request, it may help to [understand GitHub Flow](https://guides.github.com/introduction/flow/)
+
+* If you're submitting a code contribution, be sure to read the [code contributions](#code-contributions) section below.
+
+### Submitting a pull request via github.com
+
+Many small changes can be made entirely through the github.com web interface.
+
+1. Navigate to the file within [`jekyll/jekyll`](https://github.com/jekyll/jekyll) that you'd like to edit
+2. Click the pencil icon in the top right corner to edit the file
+3. Make your proposed changes
+4. Click "Propose file change"
+5. Click "Create pull request"
+6. Add a descriptive title and detailed description for your proposed change. The more information the better.
+7. Click "Create pull request"
+
+That's it! You'll be automatically subscribed to receive updates as others review your proposed change and provide feedback.
+
+### Submitting a pull request via Git command line
```

Comment:
> just to ensure that I was working off the latest commit

If we activate protected branches, we'll get a shiny "update branch" button, which auto merges in master (not to mention, Git is smart enough to do the reverse in most cases).

I'm of the opinion that there are more important things to care about than obsessing over a pristine Git history. In practice, that means it's better to have a PR with a merge commit, than not to have a PR at all because we've required the contributor (who may be submitting their first pull request) to install Git command line and learn advanced concepts like rebasing before their contributing could be considered. Practicality over purity.